google/cloud/pubsub/samples/samples.cc

```diff
@@ -218,6 +218,31 @@ void CreateSubscription(google::cloud::pubsub::SubscriptionAdminClient client,
   (std::move(client), argv.at(0), argv.at(1), argv.at(2));
 }
 
+void CreateFilteredSubscription(
+    google::cloud::pubsub::SubscriptionAdminClient client,
+    std::vector<std::string> const& argv) {
+  //! [create-filtered-subscription]
+  namespace pubsub = google::cloud::pubsub;
+  [](pubsub::SubscriptionAdminClient client, std::string const& project_id,
+     std::string const& topic_id, std::string const& subscription_id) {
+    auto sub = client.CreateSubscription(
+        pubsub::Topic(project_id, std::move(topic_id)),
+        pubsub::Subscription(project_id, std::move(subscription_id)),
+        pubsub::SubscriptionBuilder{}.set_filter(
+            R"""(attributes.is-even = "false")"""));
+    if (sub.status().code() == google::cloud::StatusCode::kAlreadyExists) {
```

Comment:
Is it preferable to make this check inside the `if (!sub) { ... }` block?

Reply:
Dunno.  As written I read the code as:

> If you get a `kAlreadyExists`, report it and return. If you get any (other) error, throw an exception. Otherwise report success.

Otherwise it reads:

> If you get any error, then check if it is `kAlreadyExists`, in that case reporting it and return, otherwise throw an exception. Otherwise report success.

The first seems slightly more readable, but I realize that might not be an argument that everybody agrees on.  FWIW, it is the pattern we use elsewhere in these examples (naturally if I am wrong that only means we have to change the pattern in more places).

Reply:
Ack on the other cases, although I would still say its less readable.  That is, I would tend to interpret a `StatusOr` as a true union/variant and never try to access `status()` until I know I'm in the error case, just like I wouldn't access `value()` until I know I'm in the non-error case.

But not a big deal, given how its implemented.  Thanks.